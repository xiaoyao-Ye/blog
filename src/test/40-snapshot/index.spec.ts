import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import Hi from './Hi.vue'

describe('40-snapshot', () => {
  it('snapshot', () => {
    expect({ name: 'Ghosteye', age: 24 }).toMatchSnapshot()
  })

  it('component', () => {
    const wrapper = mount(Hi)
    expect(wrapper.html()).toMatchSnapshot()
  })

  // 可以将快照生成为对应格式的文件, 方便高亮查看
  it('file', () => {
    const wrapper = mount(Hi)
    expect(wrapper.html()).toMatchFileSnapshot('./__snapshots__/Hi.html')
  })

  // inline snapshot 会把结果生成在当前行
  it('inline snapshot', () => {
    expect({ name: 'Ghosteye', age: 24 }).toMatchInlineSnapshot(`
      {
        "age": 24,
        "name": "Ghosteye",
      }
    `)
  })
})
