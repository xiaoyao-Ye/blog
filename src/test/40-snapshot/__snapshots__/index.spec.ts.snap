// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`40-snapshot > component 1`] = `
"<div>
  <h1 id="hi"> Hi </h1>
</div>"
`;

exports[`40-snapshot > snapshot 1`] = `
{
  "age": 24,
  "name": "Ghosteye",
}
`;
